### DISCUSSION TITLE
(WiinUSoft) Wii U Pro Controller doesn't want to pair by any chance

### DISCUSSION THREAD

**Discussion by jaydnepic** [KeyPuncher/WiinUPro#158] [Help]
I've tried for the past 2 hours straight to pair my Wii U Pro Controller thru WiinUSoft but nothing.
I've tried 6 different Bluetooth adapters, my laptop, my everything. 
And yes i do have the services installed.
![image](https://github.com/KeyPuncher/WiinUPro/assets/101939187/ae7da137-7d9b-4999-967c-b25413f3602c)

**Comment by KeyPuncher** [comment]
Well that's annoying. Is this on Windows 11?
Have you tried WiinUPro's pairing? I can't recall if both releases are exactly the same (I think they are though).

**Comment by jaydnepic** [comment]
Yes it is Windows 11, i tried WiinUPro too, nothing to do.
If it is a Windows 11 problem im resetting my pc soon so i can do it.

**Comment by KeyPuncher** [comment]
Could be a an adapter thing. My laptop still handled it with a recent update from a few days ago.
Can you pair it directly to your computer without going through WiinUSoft with an empty or 0000 pairing code?

**Comment by jaydnepic** [comment]
For the adapter it literally cant be, i tried with all of them, from the oldest bluetooth to bluetooth 5.0.
For the other pair thing im going to try it and let you know

**Comment by jaydnepic** [comment]
Ok sorry for the long wait tho i didn't have internet access for a lot of time, the controller still doesn't want to pair, screen shot below.
![image](https://github.com/KeyPuncher/WiinUPro/assets/101939187/6c647936-9bd8-47a1-ab99-da65d7373e84)

**Comment by RedDoves** [comment]
yeah, any chance here, having precisely the same problem...
really couldn't tell you if im doing anything wrong...

**Comment by KeyPuncher** [comment]
Still unsure about what's going on with this one. I've moved it over here so we can choose an answer if anyone comes across one.

**Comment by smeep0** [comment]
Tried using WiinUPro recently. Using the Sync button in the program will allow you to bypass the PIN code prompt, but now the controller just outright refuses to Sync with Windows, despite the OS itself saying the "Device is ready to be used". The lights on the controller also won't stop blinking.

Manually loading a profile will also crash the app.

**Reply by KeyPuncher** [reply]
I also just recalled something I haven't thought of in a while. Has the controller been connected to a console recently?
You may recall there was this heinous issue back in the day with controllers not behaving properly and not wanting to talk to Windows after being connected (actually I think they would disconnect after trying to open communication) where syncing the controller to a Wii U would resolve the issue.

Windows saying "Device is ready to be used" might as well be a failure. In my experience I've always had to delete that and try again.

**Reply by smeep0** [reply]
Nope. Just tried it right now by resyncing the Wii U Pro Controller to a Wii U, then attempting to let it resync with WiinUPro to no avail.

I think something in the Bluetooth stack has changed. That, and there's some XInput verification now that screws with it being recognized as a legitimate (XInput compatible) controller.

**Comment by RedDoves** [comment]
I'm out and about rn, but do you think there are different versions of Wii U pro controllers that might be causing this issue? Pure speculation. Maybe it's affecting the code somehow?

Is there any way I'd be able to run a debug for you since I've got this issue?

**Reply by KeyPuncher** [reply]
The only time I've seen issues is if they controllers aren't the official Nintendo produced controllers. The other thing that can affect this is Windows, especially Windows 11. The application hasn't been updated for Windows 11 yet.

**Reply by smeep0** [reply]
I am using an original and official Wii U Pro Controller that I have acquired way back during the console's release.

**Reply by KeyPuncher** [reply]
Same. It seems likely that it's Windows updates altering some aspect in the connection process.
